### ISSUE TITLE
CMORization of basins

### ISSUE THREAD

**Issue by pabretonniere** [EC-Earth/ece2cmor3#365]
Hi,

Both in PRIMAVERA and CMIP6 tables (Ofx table), they defined the variable basin (see definition below), but I can't figure out how to produce it in Nemo. 
Activating it as in https://dev.ec-earth.org/projects/cmip6/repository/entry/ecearth3/branches/development/2018/r5717-cmip6-nemo-namelists/runtime/classic/ctrl/cmip6-output-control-files/DCPP/cmip6-experiment-DCPP-dcppA-hindcast/file_def_nemo-opa.xml doesn't produce a variable as the  cmorized one (it has less basins), as well as the variables present in the subbasins.nc file.

Does anybody know how it can be produced/cmorized? I don't really need it the properly cmorized file but rather would want to know how the information can be obtained (for example having the  coordinates of the basins would be sufficient, but I can't find it in the tables).

Thank you

```
"basin": {
"frequency": "fx",
"modeling_realm": "ocean",
"standard_name": "region",
"units": "1",
"cell_methods": "area: mean",
"cell_measures": "area: areacello",
"long_name": "Region Selection Index",
"comment": "A variable with the standard name of region contains strings which 
indicate geographical regions. These strings must be chosen from the standard 
region list.",
"dimensions": "longitude latitude",
"out_name": "basin",
"type": "integer",
"positive": "",
"valid_min": "",
"valid_max": "",
"flag_values": "0 1 2 3 4 5 6 7 8 9 10",
"flag_meanings": "global_land southern_ocean atlantic_ocean pacific_ocean 
arctic_ocean indian_ocean mediterranean_sea black_sea hudson_bay baltic_sea 
red_sea",
"ok_min_mean_abs": "",
"ok_max_mean_abs": ""
},
```

**Comment by goord** [comment]
Hi Pierre-Antoine we only have 3 basins but I also wouldn't know where to find the corresponding masks... Perhaps a question for the shaconemo people?

**Comment by zklaus** [comment]
There should be a file `subbasins.nc` in the runtime directory of your ec-earth experiment, i.e., at the same level as the `output` directory. In there are the masks for Atlantic, Pacific, and Indian ocean, as well as Global ocean without semi-enclosed areas such as Mediterannean, Baltic, Hudson Bay.

This could be used to create a basin file, albeit incomplete. It does originally stem from shaconemo, so that could still be a good place to ask for further information.

The reason for not having the basins specified better in cmip6 is, imho, that they depend on the ocean grid. That's why the project relies solely on the name for identification, that being well standardized in the [cf convention](http://cfconventions.org/Data/standardized-region-list/standardized-region-list.html).

**Comment by zklaus** [comment]
Also, there is a [basin_mask file](https://github.com/puotila/barakuda/blob/master/data/basin_mask_ORCA1_ece3.2_2017.nc4) in barakuda that contains some more basins. Unfortunately the state of barakuda is a bit unclear, as is the applicability of that file to the current ec-earth, so care has to be taken. But from the existence of this file and the creators comment
```
[...] masks for main ocean basins, created with orca_mesh_mask_to_bitmap.py, Gimp, and tiff_to_orca_mask.py, 2017-06-02.
```
it seems like there simpy might not be an official nemo basin mask definition for all the basins.

@pabretonniere: Do you think you can construct a (static) (Ofx, basin) file from this information?

**Comment by treerink** [comment]
As the Ofx issue is solved #249 and the cited control output files above are outdated (please compare now with the `r6705-control-output-files` branch), I close this issue. Please reopen with an updated description in case this issue still needs to be addressed.
